tests/scripts/all.sh

```diff
@@ -1262,17 +1262,59 @@ component_build_psa_crypto_spm () {
     check_renamed_symbols tests/include/spe/crypto_spe.h library/libmbedcrypto.a
 }
 
-component_test_psa_crypto_client () {
-    msg "build: default config - PSA_CRYPTO_C + PSA_CRYPTO_CLIENT, make"
+# Get a list of library-wise undefined symbols and ensure that they only
+# belong to psa_xxx() functions and not to mbedtls_yyy() ones.
+# This function is a common helper used by both:
+# - component_build_default_psa_crypto_client_without_crypto_provider
+# - component_build_full_psa_crypto_client_without_crypto_provider.
+common_check_mbedtls_missing_symbols() {
+    nm library/libmbedcrypto.a | grep ' [TRrDC] ' | grep -Eo '(mbedtls_|psa_).*' | sort -u > sym_def.txt
+    nm library/libmbedcrypto.a | grep ' U ' | grep -Eo '(mbedtls_|psa_).*' | sort -u > sym_undef.txt
+    comm sym_def.txt sym_undef.txt -13 > linking_errors.txt
+    not grep mbedtls_ linking_errors.txt
+
+    rm sym_def.txt sym_undef.txt linking_errors.txt
+}
+
+component_build_default_psa_crypto_client_without_crypto_provider () {
```

Comment:
It bothers me that we're no longer running any testing with `MBEDTLS_PSA_CRYPTO_CLIENT && !MBEDTLS_PSA_CRYPTO_C`. The idea in https://github.com/Mbed-TLS/mbedtls/issues/8710#issuecomment-1933825279 was that we'd still run tests, just not doing any testing that invokes PSA functions. We do need to link programs that call PSA functions, but these functions can be stubs that just return an error. I've tried it any it works, at least, in the default config, in the full config and in the config of this component. https://github.com/gilles-peskine-arm/mbedtls/tree/issue8710-restore_non_psa_paths

Reply:
Stubs are fine for the build, but then test with opaque keys which rely on PSA fails because all the stubbed functions do not work. I think this is the last failure (hopefully) I should fix now in the CI.

I will try to play a bit with `helper_libtestdriver1_make_drivers` and `helper_libtestdriver1_make_main` to see if I can get something that we can play with more easily.

Reply:
> Stubs are fine for the build, but then test with opaque keys which rely on PSA fails because all the stubbed functions do not work.

Sure, the idea is that for now we build code with `CLIENT && !C` but we keep the tests depending on `C`. And that's only for the `!USE` build: with `USE_PSA_CRYPTO` pretty much all of pk would have to be excluded.